qiskit/algorithms/optimizers/optimizer.py

```diff
@@ -147,6 +149,24 @@ def setting(self):
         ret += f"{params}"
         return ret
 
+    @property
+    @abstractmethod
```

Comment:
I guess this is a choice here - make it abstract to force all concrete sub-classes to have an implementation. Or leave it concrete here and raise an error. Reason I bring this up is because of the global optimizers we have on nlopt. I doubt nl opt native dependent library is built here in CI so the nlopt optimizer tests I imagine are skipped. At present I do not think any will work because a `settings` is not present at least here https://github.com/Qiskit/qiskit-terra/blob/main/qiskit/algorithms/optimizers/nlopts/nloptimizer.py (or more likely all the sub-classes to make things work) as would be required if this was abstract

Reply:
Actually that still made the tests fail. But I agree it's a choice and we could also make it non-abstract. I added the settings to the nlopt optimizers for now.